src/rgw/rgw_asio_frontend.cc

```diff
@@ -411,7 +424,9 @@ int AsioFrontend::init()
 {
   boost::system::error_code ec;
   auto& config = conf->get_config_map();
-
+// Setting global timeout
+  auto timeout = config.find("request_timeout_ms");
+  request_timeout = timeout != config.end() ? std::chrono::milliseconds(std::strtoul(timeout->second.c_str(), NULL, 10)) : request_timeout;
```

Comment:
this needs error handling for the int parsing - `std::strtoul()` throws on errors. either catch them here, use one of our 'strict' helpers from common/strtol.h, or the new c++17 `std::from_chars()` (Adam added a `ceph::parse<int>()` in https://github.com/ceph/ceph/pull/33789 awaiting review, you can use that as an example)

also don't use the ternary operator here when one branch is a noop

Reply:
Fixed